src/org/labkey/test/tests/ActiveUserLimitationTest.java

```diff
@@ -0,0 +1,213 @@
+                userWarning("No").
+                setUserLimitLevel(limitLevel).
+                setUserLimitMessage("You cannot do this..! ${LimitLevel} is the limit and currently server has ${ActiveUsers} users");
+        limitActiveUserPage.save();
+
+        log("Marking previously created user as system user");
+        goToSiteUsers();
+        waitAndClickAndWait(Locator.linkWithText(USER1_DISPLAY_NAME));
+        clickButton("Edit");
+        checkCheckbox(Locator.name("quf_System"));
+        clickButton("Submit");
+
+        log("Creation of one more user should be be allowed");
+        _userHelper.createUser(USER2);
+        goToSiteUsers();
+        assertTrue("Banner should be present", isBannerPresent());
+        assertTrue("User should have been created with banner message", isTextPresent(USER2));
+    }
+
+    @Test
+    public void testActivateAndDeactivateUserWithLimits()
+    {
+        log("Creating the 2 users before the limits are set");
+        _userHelper.createUser(USER1);
+        _userHelper.createUser(USER2);
+
+        log("Setting the limit same as current active user");
+        String limitLevel = String.valueOf(getActiveUsers());
+        LimitActiveUserPage limitActiveUserPage = LimitActiveUserPage.beginAt(this);
+        limitActiveUserPage.limitActiveUsers("Yes").
+                userWarning("No").
+                setUserLimitLevel(limitLevel).
+                setUserLimitMessage("You cannot do this..! ${LimitLevel} is the limit and currently server has ${ActiveUsers} users");
+        limitActiveUserPage.save();
+
+        log("Deactivate the 2 users");
+        ShowUsersPage usersPage = goToSiteUsers();
+        usersPage.getUsersTable().setFilter("Email", "Equals One Of (example usage: a;b;c)", USER1 + ";" + USER2);
+        usersPage.getUsersTable().checkCheckbox(0);
+        usersPage.getUsersTable().checkCheckbox(1);
+        usersPage.getUsersTable().clickHeaderButton("Deactivate");
+        waitForElement(Locator.tagWithText("span", "Deactivate"));
+        clickButton("Deactivate");
+
+        log("Create another user");
+        _userHelper.createUser(USER3);
+        goToSiteUsers();
+        assertTrue("User should have been created with banner message", isTextPresent(USER3));
+
+        log("Reactivate the user");
+        usersPage = goToSiteUsers();
+        usersPage.includeInactiveUsers();
+        usersPage.getUsersTable().setFilter("Email", "Equals One Of (example usage: a;b;c)", USER1 + ";" + USER2);
+        usersPage.getUsersTable().checkCheckbox(0);
+        usersPage.getUsersTable().checkCheckbox(1);
+        usersPage.getUsersTable().clickHeaderButton("Reactivate");
+        waitForElement(Locator.tagWithText("span", "Reactivate"));
+        clickButton("Reactivate");
+        assertTrue("Missing error message", isTextPresent("Failed to activate user2: User limit has been reached so no more users can be reactivated on this deployment."));
+    }
+
```

Comment:
I think there might be a couple of other scenarios that would be worth checking. 

- Validate ActiveUserPage when invalid values are used (see comment in page file).
- Note: An invalid case would also be limit number set to a value less than warning number.
- Validate grid on ActiveUserPage (see comment in page file).
- Adding multiple users on the Add Users page (security-addUsers.view?) Add enough users so that some would be over the limit.
- What about having warning and limit both turned on and then add more users. Does the banner change?
- This maybe beyond the scope, but keep in mind the message can be html, so having something like `<script>alert("You have been <b>warned</b>! The system is limited to ${LimitLevel} users.")</script>` would be valid in the message.
- More realistically might be something like a link in the message. For example I could see this message having a link to an email, or other point of contact, to help resolve the limit issue.

I'll defer to @labkey-adam as to the value of the scenarios I described. Some of the them may already be covered in unit tests and do not need to be covered here.

Reply:
Most of them have been added as part of code review changes